cpp/src/jni/dataset/jni_util.cc

```diff
@@ -162,31 +192,40 @@ std::shared_ptr<ReservationListener> ReservationListenableMemoryPool::get_listen
 
 ReservationListenableMemoryPool::~ReservationListenableMemoryPool() {}
 
+Status CheckException(JNIEnv* env) {
+  if (env->ExceptionCheck()) {
+    env->ExceptionDescribe();
+    env->ExceptionClear();
+    return Status::Invalid("Error during calling Java code from native code");
```

Comment:
including the message with Invalid could aid users.  Status's have a member for [arbitrary details](https://github.com/apache/arrow/blob/8e43f23dcc6a9e630516228f110c48b64d13cec6/cpp/src/arrow/status.h#L101).  it might pay to create a new extension of this class that can keep a reference to the exception?

Reply:
Fixed in commit 922d6dd143c12984daa788d9d737a75914e8a678.